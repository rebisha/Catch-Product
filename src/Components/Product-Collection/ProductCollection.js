import React, { useEffect, useState } from "react";
import ProductCard from "../Product-Card/ProductCard";

import styles from "./productCollection.scss";

const ProductCollection = () => {
  const [product, setProduct] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const fetchProduct = await fetch(
          "http://catch-code-challenge.s3-website-ap-southeast-2.amazonaws.com/challenge-3/response.json"
        )
          .then(response => response.json())
          .then(data => data);

        setProduct(fetchProduct.results);
      } catch (error) {
        console.log(error);
      }
    };
    fetchData();
  }, []);

  return (
    <div className={styles.productCollection}>
      {product.map(item => (
        <ProductCard key={item.id} item={item} />
      ))}
    </div>
  );
};

export default ProductCollection;
