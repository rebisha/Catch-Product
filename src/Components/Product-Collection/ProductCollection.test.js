import React from "react";
import renderer from "react-test-renderer";
// component
import ProductCollection from "./ProductCollection";

it("Should render correctly", () => {
  const products = [
    {
      id: "ffc4211a-fb81-45e3-b1d8-2d399a92aa89",
      name: "Buy Olaplex No. 3 Hair Perfector",
      salePrice: 3145,
      retailPrice: 5000,
      imageUrl:
        "https://s.catch.com.au/images/product/0002/2114/593f690189ac9183721654_w200.jpg",
      quantityAvailable: 65,
    },
    {
      id: "46397d56-2726-45de-8514-d8ed6984a600",
      name:
        "4 x 60pk Finish Quantum Max Powerball Super Charged Dishwashing Caps Lemon Sparkle",
      salePrice: 5900,
      retailPrice: 18417,
      imageUrl:
        "https://s.catch.com.au/images/product/0001/1909/5d47c0d64988e613254534_w200.jpg",
      quantityAvailable: 56,
    },
  ];

  const wrapper = renderer.create(<ProductCollection products={products} />);
  expect(wrapper).toMatchSnapshot();
});
