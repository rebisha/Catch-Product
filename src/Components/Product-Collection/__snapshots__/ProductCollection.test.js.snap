// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<section
  className="productCollection"
>
  <div
    className="productCard"
  >
    <div
      className="productCard__wrapper"
    >
      <img
        alt="product__image"
        className="productCard__image"
        src="https://s.catch.com.au/images/product/0002/2114/593f690189ac9183721654_w200.jpg"
      />
    </div>
    <h3
      className="productCard__title"
    >
       
      Buy Olaplex No. 3 Hair Perfector
    </h3>
    <div
      className="productCard__price"
    >
      <p
        className="productCard__retailPrice"
      >
        $
        50
      </p>
      <p
        className="productCard__salePrice"
      >
         $
        31.45
         
      </p>
    </div>
  </div>
  <div
    className="productCard"
  >
    <div
      className="productCard__wrapper"
    >
      <img
        alt="product__image"
        className="productCard__image"
        src="https://s.catch.com.au/images/product/0001/1909/5d47c0d64988e613254534_w200.jpg"
      />
    </div>
    <h3
      className="productCard__title"
    >
       
      4 x 60pk Finish Quantum Max Powerball Super Charged Dishwashing Caps Lemon Sparkle
    </h3>
    <div
      className="productCard__price"
    >
      <p
        className="productCard__retailPrice"
      >
        $
        184.17
      </p>
      <p
        className="productCard__salePrice"
      >
         $
        59
         
      </p>
    </div>
  </div>
</section>
`;
